setuptools/sandbox.py

```diff
@@ -263,6 +264,52 @@ def run_setup(setup_script, args):
             # Normal exit, just return
 
 
+def _mk_dual_path_wrapper(name):
+    original = getattr(_os, name)
+
+    def wrap(self: AbstractSandbox, src, dst, *args, **kw):
+        if self._active:
+            src, dst = self._remap_pair(name, src, dst, *args, **kw)
+        return original(src, dst, *args, **kw)
+
+    return wrap
```

Comment:
I'm unsure if this refactoring is safe or why these functions were embedded. Probably it's fine to move them out, but let's do that as a separate change, either prior to adding mypy support or after by disabling the checks.

Reply:
Because they were not marked as static, the first parameter is inferred as the class' type. If you *do* type the first argument as something else w/o marking as static, you get `Self argument missing for a non-static method (or an invalid type for self)`.
In this case, these are not even used as static methods, so moving them out makes more sense.

Will revert, `type: ignore` and mark as TODO

(different, but similar set of issues in pyright)